### PR TITLE
Properly check if an identical chunk has been written before.

### PR DESCRIPTION
**Pull request by vrabaud**
Fixes internal bugs: b/321189607 and b/321189607

### CODE REVIEW

File: src/write.c [hunk 1]
```diff
@@ -1818,7 +1818,11 @@ static avifResult avifEncoderWriteMediaDataBox(avifEncoder * encoder,
             // Deduplication - See if an identical chunk to this has already been written
             if (item->encodeOutput->samples.count > 0) {
                 avifEncodeSample * sample = &item->encodeOutput->samples.sample[0];
-                chunkOffset = avifEncoderFindExistingChunk(s, mdatStartOffset, sample->data.data, sample->data.size);
+                size_t total_size = sample->data.size;
+                for (uint32_t sampleIndex = 1; sampleIndex < item->encodeOutput->samples.count; ++sampleIndex) {
+                    total_size += item->encodeOutput->samples.sample[sampleIndex].data.size;
```
Comment (y-guyon) [inline src/write.c:1823]: Is the data contiguous?

Suggestion:

```c
AVIF_CHECKERR(
  item->encodeOutput->samples.sample[sampleIndex].data.data ==
    item->encodeOutput->samples.sample[0].data.data + total_size,
  AVIF_RESULT_UNKNOWN_ERROR);
```

(not just an assertion, for extra security)

**Review by y-guyon — APPROVED** [review]
This piece of code was written 4 years ago but maybe someone more familiar with tracks than me could review.

File: src/write.c (hunk 1 above)
Comment (vrabaud) [reply src/write.c:1823]: Well, the data is written contiguously a few lines below but I'll add a check.
The problem happened starting with https://github.com/AOMediaCodec/libavif/pull/463

Comment (vrabaud) [reply src/write.c:1823]: Actually, we do not care right? The stream is already formed so as long as we have a match of size total_size, we are good and the deduplication works.

Comment (vrabaud) [reply src/write.c]: I missed it's on the input, that makes sense. Done

**Review by y-guyon — APPROVED** [review]

**Review by vigneshvg — APPROVED** [review]

**Comment by wantehchang** [comment]
@vrabaud Vincent: The first comment in this pull request says "Fixes internal bugs: b/321189607 and b/321189607", but the two bug numbers are the same. Is there a second bug that this pull request fixes?

**Comment by vrabaud** [comment]
Nope, just a mistake on my end ...

**Pull request merged**